### PR TITLE
Upload simple ci-via-container script as part of release process

### PR DESCRIPTION
**Pull request by bretambrose**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/linux-container-ci.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+BUILDER_VERSION=$1
+shift
+ECR_REPO=$1
+shift
+IMAGE_NAME=$1
+shift
+
+aws ecr get-login-password | docker login ${ECR_REPO} -u AWS --password-stdin
+export DOCKER_IMAGE=${ECR_REPO}/${IMAGE_NAME}:${BUILDER_VERSION}
+docker pull $DOCKER_IMAGE
```
Comment (justinboswell) [inline .github/workflows/linux-container-ci.sh:20]: `docker run` will pull the image automatically.

File: .github/workflows/linux-container-ci.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+docker run --env GITHUB_REF $DOCKER_IMAGE --version=${BUILDER_VERSION} build -p ${PACKAGE_NAME} "$@"
```
Comment (justinboswell) [inline .github/workflows/linux-container-ci.sh:21]: I don't think you want `$@` quoted, that'll cause it to group things together in a weird way that may not parse well

File: .github/workflows/linux-container-ci.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+# $3 - Ecr Repo
+# $4 - Image Name
+
+set -e
+
+PACKAGE_NAME=$1
+shift
+BUILDER_VERSION=$1
+shift
+ECR_REPO=$1
```
Comment (justinboswell) [inline .github/workflows/linux-container-ci.sh:13]: Does the repo ever change? Couldn't we just embed this? It's not secret really, it requires auth.

File: .github/workflows/linux-container-ci.sh (hunk 2 above)
Comment (justinboswell) [reply .github/workflows/linux-container-ci.sh:21]: Also, you could just have the args in the CI yaml start with build -p whatever whatever, so this line becomes
`docker run --env GITHUB_REF $DOCKER_IMAGE --version=${BUILDER_VERISON} $@`

File: .github/workflows/linux-container-ci.sh (hunk 3 above)
Comment (bretambrose) [reply .github/workflows/linux-container-ci.sh:13]: Will update

File: .github/workflows/linux-container-ci.sh (hunk 2 above)
Comment (bretambrose) [reply .github/workflows/linux-container-ci.sh:21]: Unsure about the quotes.  Everything I read said to use them (https://stackoverflow.com/questions/1537673/how-do-i-forward-parameters-to-other-command-in-bash-script) and a run of a multi-param invocation seems to pass them properly: https://github.com/awslabs/aws-c-io/runs/3365801050#step:2:616

Comment (bretambrose) [reply .github/workflows/linux-container-ci.sh]: Remove package_name as an arg

**Review by justinboswell — APPROVED** [review]

**Pull request merged**